libs/sdk-ui-dashboard/src/model/state/catalog/catalogSelectors.ts

```diff
@@ -0,0 +1,48 @@
+// (C) 2021 GoodData Corporation
+import { createSelector } from "@reduxjs/toolkit";
+import { DashboardState } from "../dashboardStore";
+import flatMap from "lodash/flatMap";
+
+const selectSelf = createSelector(
+    (state: DashboardState) => state,
+    (state) => state.catalog,
+);
+
+/**
+ * @internal
+ */
+export const attributesSelector = createSelector(selectSelf, (state) => {
+    return state.attributes;
+});
+
+/**
+ * @internal
+ */
+export const measuresSelector = createSelector(selectSelf, (state) => {
+    return state.measures;
```

Comment:
Will these ever get loaded? The current loadCatalog only gets dateDatasets and attributes, right? Maybe we should load everything so that the users are not confused?

Reply:
Also since the state allows undefined in these properties, the selector might be quite impractical to use, should we maybe default to an empty array?

Reply:
Good points, corrected.